### PR TITLE
Added support for new percepts and actions that have been added in the environment

### PR DESCRIPTION
**Pull request by lisv**
As a housing corporation agent I want to have access to all percepts from the environment and all indicators that will be useful to me so that I can function in the best way possible.

Most important change is that instead of all the lists of buildings, lands, zones and indicators we now insert all of them seperately (building, land, zone, indicator) so that we have easier access to them.
Also documented all code in initialize and event module and changed the code somewhat to make it better readible and more consistent.

### CODE REVIEW

**Comment by shdshdw** [comment]
After the Travis issues have been resolved this branch can be merged to dev without problems, lets do this ASAP.

File: src/nl/codefox/contextvh/tygronGoals.mod2g [hunk 1]
```diff
@@ -12,23 +12,23 @@ module tygronGoals {
...
 		then adopt(doneBuying(MultiPoly)).
 			
 	% Example of a building we will demolish	
-	if bel(building(BuildingId,'Prof. Henketstraat 48-80',_,_,_,_,_,_), ownedBuilding(BuildingId)) 
+	if bel(building(BuildingId,'Prof. Henketstraat 48-80',_,_,_,_,_,_,_), ownedBuilding(BuildingId)) 
 		then adopt( demolishBuilding(BuildingId)).
 	
 	% Example of building we will upgrade by replacing and rebuilding it
-	if bel(building(BuildingId,_,_,_,['LUXE'],FunctionID,_,MultiPolygon), ownedBuilding(BuildingId), upgrade_type(UpgradeID,FunctionID,_)) 
+	if bel(building(BuildingId,_,_,_,['LUXE'],FunctionID,_,MultiPolygon,_), ownedBuilding(BuildingId), upgrade_type(UpgradeID,UpgradeTypes), member(upgrade_pair(FunctionID,_), UpgradeTypes)  )
```
Comment (shdshdw) [inline src/nl/codefox/contextvh/tygronGoals.mod2g]: Please remove the redundant space at the end of the line.

File: src/nl/codefox/contextvh/knowledgebase.pl [hunk 2]
```diff
@@ -26,37 +31,39 @@
...
+	member(land(LandID,stakeholder(StakeholderID,_,_,_),MultiPoly,_,_),List),
+	MultiPoly\=multipolygon('MULTIPOLYGON EMPTY').
 
-% determine when a zone needs to be improved and for which indicator
-needImprovement(IndicatorID, ZoneID) :- indicator(IndicatorID, Value, Target, ZoneLink), member(zone_link(ZoneID,IndicatorID,CurrentValue,CurrentTarget), ZoneLink), CurrentValue < CurrentTarget.  
-% a zone is improved when the indicator score is higher or equal to the agent's target
-improvedZone(IndicatorID, ZoneID) :- indicator(IndicatorID, Value, Target, ZoneLink), member(zone_link(ZoneID,IndicatorID,CurrentValue,CurrentTarget), ZoneLink), CurrentValue >= CurrentTarget.
+% uses indicator scores to determine whether more Luxury houses are desirable, if this is not the case Normal houses are desirable.  
+needLuxeHouse:- indicator(34,_, _, ZoneLinkList), member(zone_link(0,_,Current1,Target1),ZoneLinkList), 
+indicator(34,_,_,ZoneLinkList), member(zone_link(1,_,Current2,Target2),ZoneLinkList),
```
Comment (shdshdw) [inline src/nl/codefox/contextvh/knowledgebase.pl]: Add tabs for proper indentation

File: src/nl/codefox/contextvh/tygronGoals.mod2g (hunk 1 above)
Comment (lisv) [inline src/nl/codefox/contextvh/tygronGoals.mod2g]: Done

File: src/nl/codefox/contextvh/knowledgebase.pl (hunk 2 above)
Comment (lisv) [inline src/nl/codefox/contextvh/knowledgebase.pl]: What do you mean exactly? Why and where should there be tabs?

File: src/nl/codefox/contextvh/tygronEvents.mod2g [hunk 3]
```diff
@@ -4,51 +4,71 @@ use tygronGoals as module.
...
+	% insert new lands
+	forall percept(lands(List)), bel(member(land(LandID,OwnerID,MultiPoly,ZoneIDList,Size),List), not(land(LandID,_,_,_,_)))
+		do insert(land(LandID,OwnerID,MultiPoly,ZoneIDList,Size)).
+	% delete old lands that no longer exist
+	forall percept(lands(List)), bel(land(LandID,OwnerID,OldMultiPoly,OldZoneList,OldSize), not(member(land(LandID,_,_,_,_),List)))
+		do delete(land(LandID,OwnerID,OldMultiPoly,OldZoneList,OldSize)).
+
+	%%% INDICATOR PERCEPTS
+	% update indicator/4 [indicator(IndicatorID,Current,Target,ZoneIndicatorsList)]]
+	%	when another percept is received with the same IndicatorID but with other values.
```
Comment (MHMCD) [inline src/nl/codefox/contextvh/tygronEvents.mod2g]: this should be a space after the `%` instead of a tab.

Comment (lisv) [inline src/nl/codefox/contextvh/tygronEvents.mod2g]: Okay I adjusted this.

**Comment by shdshdw** [comment]
All issues have been resolved, ready for merge.

**Comment by MHMCD** [comment]
Looks good now, is ready to be merged.

**Comment by MatthijsJWH** [comment]
To be honest, I don't like the removing of spaces, it has the tendency to make the code less readable.

**Comment by MatthijsJWH** [comment]
Other than that, everything looks fine to me.

**Pull request merged**